### ISSUE TITLE
False positive triggers for `async_without_await` in `actor` with `init() async`

### ISSUE THREAD

**Issue by akuzminskyi** [realm/SwiftLint#6423]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

In the following example, `async_without_await` suggest to remove `async` from `init` method, but it causes an error. 
```
actor A {
    init() async {
        foo()
    }

    func foo() {}
}
```

<img width="647" height="139" alt="Image" src="https://github.com/user-attachments/assets/b96b40a9-d4d8-4cdb-89ea-9e84365a2eff" />

This is only for `actor` as we can write without `await`, so it doesn't affect other data types. 

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version: 0.63.0
* Xcode version: 26.0.1
* Installation method used: [Universal macOS Binary](https://github.com/realm/SwiftLint/releases/download/0.63.0/portable_swiftlint.zip)
* Configuration file:

```yml
only_rules:
- async_without_await
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

**Comment by SimplyDanny** [comment]
It's probably best to just ignore initializers of actors from the analysis.
